### PR TITLE
In SelectMulti, determine currently selected options by *value*, rather than by *reference*

### LINKED ISSUE #78: In SelectMulti (& confirm in SelectSingle), determine currently selected options by *value*, rather than by *reference*
Currently, `SelectMulti` uses a simple `indexOf` to determine if a given option is currently selected.

It does this by checking "is this item from the `options` array, present in the current `selectedOptions`?"

See here: https://github.com/politico/vue-accessible-selects/blob/v0.12.0/src/SelectMulti.vue#L272
```ts
const optionIndex = this.selectedOptions.indexOf(option)
```

Or here: https://github.com/politico/vue-accessible-selects/blob/v0.12.0/src/SelectMulti.vue#L356
```ts
'option-selected': selectedOptions.indexOf(option) > -1,
```

This works fine for most cases: we expect that the `options` passed into the component will remain the same as when the component was initialized, and as such, it should be safe to *compare by reference*, as `indexOf` does, to see "is this specific object reference in the selected array"

A problem occurs, however, as soon as we start to have more complicated parent components, that might update the list of `options` passed into `SelectMulti` - perhaps based on an API call. Now, even though *the original values* of the `options` list may still exist, they now *have a new reference*. And thus, `indexOf` is no longer an accurate way to find these options & determine if they are in the `selectedOptions` array

**Solution:**
Allow for "find by value", rather than "find by reference". i.e. let's choose a way to reliably compare the items in the `options` array, with those that have been added to the `selectedOptions` array. That *does not* rely on objects in each array having the same reference. Since the reference approach won't be terribly maintainable, and will quickly lead to the types of confusing bugs we've already begun to experience @ Politico.

In this issue, then, let's first write a failing test:
"When I update the `options` prop to be an array full of new references, selected options are still correctly displayed as long as they match the value of current `selectedOptions`"

Then, let's fix the code so the spec passes

*Note: `SelectSingle` is already doing compare-by-value [here](https://github.com/politico/vue-accessible-selects/blob/v0.12.0/src/SelectSingle.vue#L70), but let's make sure it has the appropriate specs & testing regardless*

### PR DESCRIPTION
**Pull request by jackkoppa**
Closes #78

### CODE REVIEW

**Comment by jackkoppa** [comment]
**Note: not ready to merge yet: first, we want to confirm spec failures**

**Comment by jackkoppa** [comment]
Yay! We're currently at a great place, with 2 types of failing specs that need to be fixed before this PR is opened for review

## Need to resolve

1. `SelectMulti` - specs are failing because of the issue described in #78  - an options array updated with new references to the same values isn't being respected when checking which options should be treated as selected
2. `SelectSingle` - refactoring the specs here & adding a couple pointed out an important bug: the item in the dropdown that's marked as selected is not correct, **when using `uniqueIdField`**

**Comment by jackkoppa** [comment]
Yay! Confirmed that this commit: https://github.com/politico/vue-accessible-selects/pull/79/commits/ead3673b70c72f6fa59d9b8bf7d26aad2212e3e9

Resolves the issue 1 - the `SelectMulti` compare-by-value issue 

See test log: https://github.com/politico/vue-accessible-selects/pull/79/checks?check_run_id=1948305747

**Comment by jackkoppa** [comment]
And yay! This commit: https://github.com/politico/vue-accessible-selects/pull/79/commits/d1f7514726aaf93b468592b75ca9fbc3a1b9a84f

Resolves issue 2 - the `SelectSingle` select option highlighting bug

See [test log](https://github.com/politico/vue-accessible-selects/pull/79/checks?check_run_id=1948392762)

**Comment by jackkoppa** [comment]
PR now ready for review

**Comment by jackkoppa** [comment]
## Note: should not be merged until we do some testing in our own projects, to verify that this is a problem worth solving

File: src/SelectSingle.vue [hunk 1]
```diff
@@ -105,16 +108,32 @@
 				return this.disabled || this.loading
 			},
 			selectedIndex(): number {
-				return this.value
-				? this.options.findIndex(currentOption => currentOption.value == this.value.value)
```
Comment (jackkoppa) [inline src/SelectSingle.vue:109]: This was the actual bug in SelectSingle re: which option was showing as selected in the dropdown, when using `uniqueIdField`

File: src/SelectSingle.vue [hunk 2]
```diff
@@ -216,14 +235,13 @@
 			onOptionChange(index: number) {
 				this.activeIndex = index
 			},
+			optionExists(option: SelectOption) {
```
Comment (EvanSanderson) [inline src/SelectSingle.vue:238]: Do we need this method? Since it invokes and returns another helper function, could we just cut out the middle man, so to speak?

File: specs/SelectMulti.spec.ts [hunk 3]
```diff
@@ -12,16 +12,80 @@ describe('SelectMulti', () => {
...
+				const listbox = wrapper.find("div[role=listbox]")
+				const selectedDivOptions = listbox.findAll('div.option-selected[role=option][aria-selected=true]')
+
+				expect(selectedDivOptions.at(0).text()).toEqual('Item Two')
+				expect(selectedDivOptions.at(1).text()).toEqual('Item Three')
+			})
+
+			testBehaviorForEventEmitsAndVModel?.()
+
+			describe('(testing finding items by value, rather than reference) ' +
```
Comment (EvanSanderson) [inline specs/SelectMulti.spec.ts:53]: This describe text is a little hard to follow - I'm following whats happening up until this point and then it I'm like 🤪 . But maybe thats just because this is a complicated use case? Thoughts?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -5,11 +5,12 @@ name: Verify
 
 # Controls when the action will run. 
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
+  # Triggers the workflow on either push to main, or pull requests to any branch
```
Comment (EvanSanderson) [inline .github/workflows/test.yml:8]: Do we need to indicate on readme this change, so contributors are aware? Or does that not matter?

**Review by EvanSanderson — COMMENTED** [review]
👍 see comments beloowwww

File: specs/SelectMulti.spec.ts (hunk 3 above)
Comment (jackkoppa) [reply specs/SelectMulti.spec.ts:53]: Yeah, let me take another shot at it. It's complicated, but it's important for it to make sense to future people

File: .github/workflows/test.yml (hunk 4 above)
Comment (jackkoppa) [reply .github/workflows/test.yml:8]: I... don't think so. I think if we setup the Actions "well enough", they should appear to "just work"

Contributors shouldn't have to worry about them until or unless they fail, and so I don't think it's worth being in the README. However, we do still have an issue open to add a CONTRIBUTING.md, and while info about Actions may or may not be necessary, things about linting or commit naming are definitely worthwhile - see #63

File: src/SelectSingle.vue (hunk 2 above)
Comment (jackkoppa) [reply src/SelectSingle.vue:238]: It might over-complicate things! It makes it a *little* easier to write when being used (we're always going to have the same `this.uniqueIdField` when in the component, so we don't need to pass it in each time), but I think you're right - let's just add the helper to the instance & call it directly

**Review by ikniaziuk — APPROVED** [review]

**Review by ee923925github — APPROVED** [review]

**Comment by jackkoppa** [comment]
Howdy @wickkidd et al. - I had created this PR right around the time I was finishing up at Politico. It was... maybe helpful at the time, but now that I'm looking at my GitHub PRs more regularly again, I'm noticing it was never merged. Given I assume it's not causing an active issue for you, I think I'll go ahead & close to not break anything - the commits will still be around for future reference, obviously, in case something similar ends up being useful later. Doing the same for this other stale PR:

https://github.com/politico/vue-accessible-selects/pull/77
